Moving/Updating the WasmPlugin tutorial and Adding ImagePullPolicy description (#11379)
istio/istio.io: content/en/docs/tasks/extensibility/wasm-module-distribution/index.md

These extensions must first be distributed to the Envoy proxy.
Istio makes this possible by allowing the proxy agent to dynamically download Wasm modules.

## Setup the Test Application

Before you begin this task, please deploy the [Bookinfo](/docs/examples/bookinfo/#deploying-the-application) sample application.

## Configure Wasm Modules

In this example, you will add a HTTP Basic auth extension to your mesh. You will configure Istio to pull the [Basic auth module](https://github.com/istio-ecosystem/wasm-extensions/tree/master/extensions/basic_auth) from a remote image registry and load it. It will be configured to run on calls to `/productpage`.

To configure a WebAssembly filter with a remote Wasm module, create a `WasmPlugin` resource:

{{< text bash >}}

> $ kubectl apply -f - <<EOF

Reviewer: Is this format required for doc tests?

Author: Yes.

        credentials:
          - "ok:test"
          - "YWRtaW4zOmFkbWluMw=="
EOF
{{< /text >}}

An HTTP filter will be injected into ingress gateway proxies as an authentication filter.
The Istio agent will interpret the `WasmPlugin` configuration, download remote Wasm modules from the OCI image registry to a local file, and inject the HTTP filter into Envoy by referencing that file.

Please note that if a `WasmPlugin` is created in a specific namespace besides `istio-system`, the pods in the namespace will be configured. In case of `istio-system` namespace, all the namespace will be affected.

## Check the configured Wasm module

1. Test `/productpage` without credential

{{< text bash >}}
$ curl -s -o /dev/null -w "%{http_code}" "http://$INGRESS_HOST:$INGRESS_PORT/productpage"
401
{{< /text >}}

1. Test `/productpage` with credential

{{< text bash >}}
$ curl -s -o /dev/null -w "%{http_code}" -H "Authorization: Basic YWRtaW4zOmFkbWluMw==" "http://$INGRESS_HOST:$INGRESS_PORT/productpage"
200
{{< /text >}}

For more example usage of the `WasmPlugin` API, please take a look at the [API reference](/docs/reference/config/proxy_extensions/wasm-plugin/).

> ## Cleanup Wasm Modules

Reviewer: ```suggestion
## Clean up Wasm modules
```

        credentials:
          - "ok:test"
          - "YWRtaW4zOmFkbWluMw=="
EOF
{{< /text >}}

An HTTP filter will be injected into ingress gateway proxies as an authentication filter.
The Istio agent will interpret the `WasmPlugin` configuration, download remote Wasm modules from the OCI image registry to a local file, and inject the HTTP filter into Envoy by referencing that file.

> Please note that if a `WasmPlugin` is created in a specific namespace besides `istio-system`, the pods in the namespace will be configured. In case of `istio-system` namespace, all the namespace will be affected.

Reviewer: ```suggestion
{{< idea >}}
If a `WasmPlugin` is created in a specific namespace besides `istio-system`, the pods in that namespace will be configured. If the resource is created in the `istio-system` namespace, all namespaces will be affected.
{{< /idea >}}
```

## Check the configured Wasm module

> 1. Test `/productpage` without credential

Reviewer: ```suggestion
1. Test `/productpage` without credentials
```

{{< text bash >}}
$ curl -s -o /dev/null -w "%{http_code}" "http://$INGRESS_HOST:$INGRESS_PORT/productpage"
401
{{< /text >}}

> 1. Test `/productpage` with credential

Reviewer: ```suggestion
1. Test `/productpage` with credentials
```
